### ISSUE TITLE
Capture variables must be lower-case

### ISSUE THREAD

**Issue by ncannasse** [HaxeFoundation/haxe#5589]
I find this error to be quite misleading, for instance in the following case:

``` haxe
enum E {
   A;
   B;
   C;
}

var v = A;
switch( v ) {
case D: // Capture variables must be lower-case
}
```

A more appropriate error would be `D is not a constructor for enum E`, at least when matching an enum.

**Comment by ncannasse** [comment]
@Simn Could we have this fixed for next RC?

**Comment by Simn** [comment]
Too risky, identifier handling in patterns is bad enough as it is.

**Comment by Simn** [comment]
This will be addressed by #6207.
